### PR TITLE
chromium: update to 91.0.4472.77

### PR DESCRIPTION
**Pull request by msisov**
Stable release announcement:
https://chromereleases.googleblog.com/2021/05/stable-channel-update-for-desktop_25.html

License changes
    ---------------
    Added licenses:
      * third_party/highway/LICENSE
      * third_party/libjxl/LICENSE
      - Add libjxl library and its dependencies.
      - https://chromium-review.googlesource.com/c/chromium/src/+/2745343

      * third_party/setupdesign/LICENSE
      - third_party: Added setupdesign library LICENSE file
      - https://chromium-review.googlesource.com/c/chromium/src/+/2745578

    Removed licenses:
      * third_party/schema_org/LICENSE
      - Remove media feed and support code
      - https://chromium-review.googlesource.com/c/chromium/src/+/2803838

Fixes: 515

### CODE REVIEW

File: meta-chromium/recipes-browser/chromium/chromium-gn.inc [hunk 1]
```diff
@@ -54,6 +55,7 @@ DEPENDS += " \
     atk \
     bison-native \
     cairo \
+    curl \
```
Comment (rakuco) [inline meta-chromium/recipes-browser/chromium/chromium-gn.inc:58]: Does the build fail without curl? I remember seeing some changes related to using Crashpad with libcurl, but we don't really support Crashpad here, right?

File: meta-chromium/recipes-browser/chromium/files/0001-Remove-the-GN-settings-done-for-clang-that-conflict-.patch [hunk 2]
```diff
@@ -1,6 +1,6 @@
-From cab3c06864569577158626e500b2253687fc09ba Mon Sep 17 00:00:00 2001
-From: Khem Raj <raj.khem@gmail.com>
-Date: Mon, 29 Apr 2019 12:00:19 +0300
+From 5e5a15213a1a90f45fab8bd13fbdaabfeea8e20c Mon Sep 17 00:00:00 2001
+From: Maksim Sisov <msisov@igalia.com>
+Date: Thu, 3 Jun 2021 08:57:46 +0300
```
Comment (rakuco) [inline meta-chromium/recipes-browser/chromium/files/0001-Remove-the-GN-settings-done-for-clang-that-conflict-.patch:3]: Hmm, this change looks odd.

File: meta-chromium/recipes-browser/chromium/chromium-gn.inc (hunk 1 above)
Comment (msisov) [reply meta-chromium/recipes-browser/chromium/chromium-gn.inc:58]: That's the problem -

```
| arm-poky-linux-gnueabi-clang++ -target arm-poky-linux-gnueabi   -mfpu=neon-vfpv4 -mfloat-abi=hard -mcpu=cortex-a7 -rtlib=compiler-rt --unwindlib=libgcc --stdlib=libc++ -mlittle-endian -Qunused-arguments -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0/recipe-sysroot -MMD -MF obj/third_party/crashpad/crashpad/util/util/http_transport_libcurl.o.d -DZLIB_CONST -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DOFFICIAL_BUILD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNO_UNWIND_TABLES -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DCRASHPAD_ZLIB_SOURCE_EXTERNAL -DCRASHPAD_LSS_SOURCE_EXTERNAL -Igen/third_party/crashpad/crashpad -I../.. -Igen -I../../third_party/crashpad/crashpad -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -I../../third_party/crashpad/crashpad/compat/linux -I../../third_party/crashpad/crashpad/compat/non_win -I../../third_party/abseil-cpp -I../../third_party/boringssl/src/include -I../../third_party/protobuf/src -Igen/protoc_out -I../../third_party/zlib -fno-delete-null-pointer-checks -fno-ident -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -flto=thin -fsplit-lto-unit -fwhole-program-vtables -Xclang -fdebug-compilation-dir -Xclang . -mfpu=neon-vfpv4 -mthumb -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-unknown-warning-option -Wno-psabi -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-non-c-typedef-for-linkage -Wno-max-tokens -O2 -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g0 -ftrivial-auto-var-init=pattern -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -I/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0/recipe-sysroot/usr/include/glib-2.0 -I/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0/recipe-sysroot/usr/lib/glib-2.0/include -DPROTOBUF_ALLOW_DEPRECATED=1 -std=c++14 -fno-trigraphs -Wno-trigraphs -fno-exceptions -fno-rtti -fvisibility-inlines-hidden  -O2 -pipe  -feliminate-unused-debug-types -fmacro-prefix-map=/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0=/usr/src/debug/chromium-ozone-wayland/91.0.4472.77-r0                      -fdebug-prefix-map=/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0=/usr/src/debug/chromium-ozone-wayland/91.0.4472.77-r0                      -fdebug-prefix-map=/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/gitlab-runner/yocto-chromium/builds/ci-chromium-wayland-raspberrypi3-clang-gatesgarth/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/chromium-ozone-wayland/91.0.4472.77-r0/recipe-sysroot-native= -g1 --stdlib=libc++ -c ../../third_party/crashpad/crashpad/util/net/http_transport_libcurl.cc -o obj/third_party/crashpad/crashpad/util/util/http_transport_libcurl.o
| ../../third_party/crashpad/crashpad/util/net/http_transport_libcurl.cc:17:10: fatal error: 'curl/curl.h' file not found
| #include <curl/curl.h>
|          ^~~~~~~~~~~~~
```

I didn't see we ever tried to disable crashpad.

File: meta-chromium/recipes-browser/chromium/files/0001-Remove-the-GN-settings-done-for-clang-that-conflict-.patch (hunk 2 above)
Comment (msisov) [reply meta-chromium/recipes-browser/chromium/files/0001-Remove-the-GN-settings-done-for-clang-that-conflict-.patch]: oopps. fixed

File: meta-chromium/recipes-browser/chromium/chromium-gn.inc (hunk 1 above)
Comment (rakuco) [reply meta-chromium/recipes-browser/chromium/chromium-gn.inc:58]: I meant more in the sense that we don't have anything in place for uploading the crashes or anything. But if the build fails without curl, then let's keep it as a dependency.

**Review by rakuco — APPROVED** [review]
lgtm

**Comment by msisov** [comment]
Thanks

**Pull request merged**